### ISSUE TITLE
Support address bulk aliases

### ISSUE THREAD

**Issue by sizgiyaev** [netascode/terraform-provider-ipam#12]
Hi,
There are cases where we need to maintain ipam for more than one subnet, while the pool of the IPs should be created and named using interpolation. It is impossible today as the entire provider block could not be handled with `for_each`, `count`, etc... 
I would suggest adding support for aliases for address blocks.
Example:
```
provider "ipam" {
  addresses = {
     vlan1000 = [
       {      
          ip            = "10.10.10.1"
          prefix_length = "24"
          gateway       = "10.10.10.254"
       },
       {      
          ip            = "10.10.10.2"
          prefix_length = "24"
          gateway       = "10.10.10.254"
       },
       {      
          ip            = "10.10.10.3"
          prefix_length = "24"
          gateway       = "10.10.10.254"
       }
     ],
     vlan1001 = [
       {
          ip            = "10.10.11.1"
          prefix_length = "24"
          gateway       = "10.10.11.254"
        },
       {
          ip            = "10.10.11.2"
          prefix_length = "24"
          gateway       = "10.10.11.254"
        },
       {
          ip            = "10.10.11.3"
          prefix_length = "24"
          gateway       = "10.10.11.254"
        }
     ]
}

resource "ipam_allocate" "vlan1000_hosts" {
   block = "vlan1000"
   addresses = {
    "host1" = {}
    "host2" = {}
  }
}

resource "ipam_allocate" "vlan1001_hosts" {
   block = "vlan1001"
   addresses = {
    "host3" = {}
    "host4" = {}
  }
}
```

**Comment by danischm** [comment]
As a workaround you could use provider aliases (https://www.terraform.io/language/providers/configuration#alias-multiple-provider-configurations) to define multiple configurations (blocks) and then refer to them on a per-resource basis. Like the following:

```terraform
provider "ipam" {
  alias    = "vlan1000"
  addresses = [
    {      
      ip            = "10.10.10.1"
      prefix_length = "24"
      gateway       = "10.10.10.254"
    },
    {      
      ip            = "10.10.10.2"
      prefix_length = "24"
      gateway       = "10.10.10.254"
    },
    {      
      ip            = "10.10.10.3"
      prefix_length = "24"
      gateway       = "10.10.10.254"
    }
  ]
}

provider "ipam" {
  alias    = "vlan1001"
  addresses = [
    {      
      ip            = "10.10.10.1"
      prefix_length = "24"
      gateway       = "10.10.10.254"
    },
    {      
      ip            = "10.10.10.2"
      prefix_length = "24"
      gateway       = "10.10.10.254"
    },
    {      
      ip            = "10.10.10.3"
      prefix_length = "24"
      gateway       = "10.10.10.254"
    }
  ]
}

resource "ipam_allocate" "vlan1000_hosts" {
  provider  = "vlan1000"
  addresses = {
    "host1" = {}
    "host2" = {}
  }
}

resource "ipam_allocate" "vlan1001_hosts" {
  provider = "vlan1001"
  addresses = {
    "host3" = {}
    "host4" = {}
  }
}
```

Though this approach has some limitations as well, as the provider alias must be more or less static and cannot be defined dynamically like using variable interpolation.

**Comment by sizgiyaev** [comment]
The problem is that I cannot predict the names of aliases, it should be dynamic for our needs.  Unfortunately terraform doesn't support interpolated providers alias names

**Comment by danischm** [comment]
This should be addressed in v0.1.0 of the provider.
